theano/sandbox/cuda/tests/test_opt.py

```diff
@@ -56,6 +56,15 @@ def test_local_assert():
     assert isinstance(a_op[0].inputs[0].type, CudaNdarrayType)
 
 
+def test_local_remove_all_assert():
+    x = theano.tensor.fmatrix()
+    a = theano.tensor.opt.assert_op(x, theano.tensor.eq(x, 0).any())
+    f = theano.function([x], a, mode=mode_with_gpu)
+    topo = f.maker.fgraph.toposort()
+    a_op = [n for n in topo if isinstance(n.op, theano.tensor.opt.Assert)]
+    assert len(a_op) == 0
```

Comment:
This test should fail, for the simple raise that "unsafe" wasn't added to the list of tag to use when selecting optimization.

do
mode=mode_with_gpu.including('unsafe')
and use this mode.

Did you ran this tests? If it was passing, it mean there was a problem.